### PR TITLE
Create params structs to encapsulate version setting

### PR DESCRIPTION
**Pull request by stevenjack**
### Problem

We're calling the API without a version number in batch calls, we shouldn't be doing this.

### Solution

Refactor the params usage to have a single struct for `BatchParams` that adds the version number to any relative url.

### CODE REVIEW

**Comment by stevenjack** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;sTxfROgM&#x2F;331-fbi-set-api-version-for-batch-calls"> [FBI] Set API version for batch calls</a>

File: params.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+		Method      string
+		RelativeUrl string
+	}
+)
+
+func NewParams(endpoint string, params map[string]interface{}) Params {
+	return Params{endpoint, params}
+}
+
+func NewBatchParams(endpoint string) BatchParams {
```
Comment (revett) [inline params.go]: 🏄 comment

File: params.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	}
+
+	// BatchParams comment pending
+	BatchParams struct {
+		Method      string
+		RelativeUrl string
+	}
+)
+
+func NewParams(endpoint string, params map[string]interface{}) Params {
```
Comment (revett) [inline params.go]: 🏄 comment

File: params.go [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+	return Params{endpoint, params}
+}
+
+func NewBatchParams(endpoint string) BatchParams {
+	return BatchParams{RelativeUrl: endpoint}
+}
+
+func (p BatchParams) ToFBParams() facebookLib.Params {
```
Comment (revett) [inline params.go]: 🏄 comment

Comment (revett) [inline params.go]:
```
ToFbParams
```

🏄

File: params.go [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+	p.RelativeUrl = fmt.Sprintf("%s/%s", facebookAPIVersion, p.RelativeUrl)
+
+	return facebookLib.Params{
+		"method":       p.Method,
+		"relative_url": p.RelativeUrl,
+	}
+}
+
+func (p Params) ToFBParams() facebookLib.Params {
```
Comment (revett) [inline params.go]: 🏄 comment

Comment (revett) [inline params.go]: Camel case `Fb` 🏄

File: params.go [hunk 5]
```diff
@@ -0,0 +1,44 @@
+package fbintegration
+
+import (
+	"fmt"
+	facebookLib "github.com/huandu/facebook"
+)
+
+type (
+	Params struct {
```
Comment (revett) [inline params.go]: 🏄 comment

**Comment by revett** [comment]
🍰 after surfers

**Pull request merged**